src/split_single_dyn_dim.cpp

```diff
@@ -36,36 +36,81 @@ inline namespace MIGRAPHX_INLINE_NS {
 struct dynamic_dimensions_check
 {
     std::string dyn_param_str;
-    size_t dyn_index;
-    size_t min_dim;
-    size_t max_dim;
+    shape::dynamic_dimension dd;
 };
 
-optional<dynamic_dimensions_check>
-has_one_dyn_dim(const std::unordered_map<std::string, shape>& param_shapes)
+/**
+ * Returns value if the parameters contain non-fixed dynamic_dimensions that are the same between
+ * all of the dynamic shape parameters.
+ * In other words, each parameter can have one non-fixed dynamic_dimension `x` where `x` is the same
+ * between all of the parameters with a non-fixed dynamic_dimension.
+ * Returns the parameters and the dynamic dimension in a vector of dynamic_dimensions_check objects.
+ */
+optional<std::vector<dynamic_dimensions_check>>
+has_one_unique_dyn_dim(const std::unordered_map<std::string, shape>& param_shapes)
 {
-    // True if parameters contain exactly one dynamic shape with exactly one non-fixed
-    // dynamic_dimension.
+    if(param_shapes.empty())
```

Comment:
This special check isn't required. The return at line 60 should do just fine, without this clause.

Reply:
or vice versa.